### PR TITLE
    Port #7925 and #8365 from core strategies to parent_select plugin

### PR DESCRIPTION
**Pull request by rob05c**
Ports #7925 and #8365 from core strategies to parent_select plugin.
    
PR #7925 adds go_direct support to peering_ring strategies.
PR #8365 refactors strategy consistent hashing to fix a bug.
    
They both essentially completely rewrite the strategy consistent
hashing, so it was safer and easier to port them both at the same
time.

This PR is based on https://github.com/apache/trafficserver/pull/8580 - recommend merging that first.

### CODE REVIEW

**Comment by rob05c** [comment]
Please don't merge yet. This is a significant refactor, I still need to manually test the features as much as possible to ensure it all works right and was ported correctly.

It's not a Draft to make the tests run. I'll be reverting to a draft as soon as the tests finish.

**Comment by ywkaras** [comment]
Can you, as a part of this PR, create versions of the Au tests zzz_strategies_peer.test.py and zzz_strategies_peer2.test.py that utilize the plugin rather than strategies.yaml?

**Comment by rob05c** [comment]
> Can you, as a part of this PR, create versions of the Au tests zzz_strategies_peer.test.py and zzz_strategies_peer2.test.py that utilize the plugin rather than strategies.yaml?

Done.

**Review by ywkaras — APPROVED** [review]

File: plugins/experimental/parent_select/consistenthash.cc [hunk 1]
```diff
@@ -297,6 +320,8 @@ PLNextHopConsistentHash::next(TSHttpTxn txnp, void *strategyTxn, const char *exc
 
   PL_NH_Debug(PL_NH_DEBUG_TAG, "nextParent NH plugin calling");
 
+  uint32_t const NO_RING_USE_POST_REMAP = uint32_t(0) - 1;
```
Comment (ywkaras) [inline plugins/experimental/parent_select/consistenthash.cc:323]: Constructor-style casts are not Leif-legal but hopefully he won't see it.

**Review by ywkaras — APPROVED** [review]

File: plugins/experimental/parent_select/consistenthash.cc (hunk 1 above)
Comment (rob05c) [reply plugins/experimental/parent_select/consistenthash.cc:323]: I didn't write this, this PR is just porting the existing code/changes in core to the plugin
https://github.com/apache/trafficserver/blob/3e8f2eba/proxy/http/remap/NextHopConsistentHash.cc#L235

**Comment by zwoop** [comment]
Cherry-picked to v9.2.x

**Pull request merged**